odl/tomo/data/uncompr_bin.py

```diff
@@ -0,0 +1,472 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Utilities and class for reading uncompressed binary files with header."""
+
+import csv
+import numpy as np
+import struct
+
+
+__all__ = ('FileReaderUncompressedBinary',)
+
+
+# TODO: add rest
+DTYPE_MAP_NPY2STRUCT = {
+    np.dtype('float32'): 'f',
+    np.dtype('int32'): 'i',
+    np.dtype('int16'): 'h',
+    np.dtype('int8'): 'b',
+    np.dtype('uint64'): 'L',
+    np.dtype('uint32'): 'I',
+    np.dtype('uint16'): 'H',
+    np.dtype('uint8'): 'B',
+    np.dtype('S1'): 'b'}
+
+
+def reformat_spec(spec):
+    """Return the reformatted specification table.
+
+    The given specification is assumed to be in
+    `reST grid table format
+    <http://docutils.sourceforge.net/docs/user/rst/quickref.html#tables>`_.
+
+    Parameters
+    ----------
+    spec : str
+        Specification given as a string containing a definition table.
+
+    Returns
+    -------
+    lines : tuple of strings
+        Table lines with leading and trailing '|' stripped and lines
+        containing '...' removed.
+    """
+    return tuple(line[1:-1].rstrip() for line in spec.splitlines()
+                 if line.startswith('|') and '...' not in line)
```

Comment:
This function is used only once in the `spec_fields` function. It's simple enough to be just inlined.